### ISSUE TITLE
gitextract Illegal characters in path

### ISSUE THREAD

**Issue by roryclaasen** [ericlemes/vcsparser#19]
gitextract crashes when running against [WebKit](https://github.com/WebKit/webkit).

## Repro

1. Clone [WebKit/webkit.git](https://github.com/WebKit/webkit)

2. Run `gitextract` on cloned repo

    ```shell
    $ vcsparser gitextract --gitlogcommand "git log --pretty=fuller --date=iso --after=2018-04-07 --numstat" --output "codechurn" --output-type "MultipleFile" --bugregexes "(?i)((?:^|\W)bug(?:$|\W)|(?:^|\W)crash(?:$|\W)|(?:^|\W)crashes(?:$|\W))|(?:^|\W)bugs(?:$|\W)|(?:^|\W)bugfix(?:$|\W)|(?:^|\W)bugfixes(?:$|\W)"
    ```

## Stacktrace

```
Unhandled Exception: Newtonsoft.Json.JsonSerializationException: Error getting value from 'Extension' on 'vcsparser.core.DailyCodeChurn'. ---> System.ArgumentException: Illegal characters in path.
   at System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
   at System.IO.Path.GetExtension(String path)
   at vcsparser.core.DailyCodeChurn.get_Extension() in E:\Source\repos\vcsparser\src\vcsparser.core\DailyCodeChurn.cs:line 31
   at GetExtension(Object )
   at Newtonsoft.Json.Serialization.DynamicValueProvider.GetValue(Object target)
   --- End of inner exception stack trace ---
   at Newtonsoft.Json.Serialization.DynamicValueProvider.GetValue(Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.CalculatePropertyValues(JsonWriter writer, Object value, JsonContainerContract contract, JsonProperty member, JsonProperty property, JsonContract& memberContract, Object& memberValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Serialize(TextWriter textWriter, Object value)
   at vcsparser.core.OutputProcessor.ProcessOutputSingleFile[T](String fileName, IList`1 result) in E:\Source\repos\vcsparser\src\vcsparser.core\OutputProcessor.cs:line 34
   at vcsparser.core.OutputProcessor.ProcessOutputMultipleFile[T](String filePrefix, Dictionary`2 dict) in E:\Source\repos\vcsparser\src\vcsparser.core\OutputProcessor.cs:line 112
   at vcsparser.core.OutputProcessor.ProcessOutput[T](OutputType outputType, String outputFile, Dictionary`2 dict) in E:\Source\repos\vcsparser\src\vcsparser.core\OutputProcessor.cs:line 126

   at vcsparser.core.git.GitCodeChurnProcessor.Extract() in E:\Source\repos\vcsparser\src\vcsparser.core\git\GitCodeChurnProcessor.cs:line 72
   at vcsparser.Program.RunGitCodeChurnProcessor(GitExtractCommandLineArgs a) in E:\Source\repos\vcsparser\src\vcsparser\Program.cs:line 73
   at vcsparser.Program.<>c.<Main>b__0_2(GitExtractCommandLineArgs a) in E:\Source\repos\vcsparser\src\vcsparser\Program.cs:line 27
   at CommandLine.ParserResultExtensions.MapResult[T1,T2,T3,TResult](ParserResult`1 result, Func`2 parsedFunc1, Func`2 parsedFunc2, Func`2 parsedFunc3, Func`2 notParsedFunc)
   at vcsparser.Program.Main(String[] args) in E:\Source\repos\vcsparser\src\vcsparser\Program.cs:line 24
```

**Comment by roryclaasen** [comment]
Solution is to set `quotepath` to `off` in gitconfig

```shell
git config core.quotepath off
```

**Comment by roryclaasen** [comment]
Fixed in 49eda116ab0b0231795c86620b6be9e878a53729
